functional/pipeline.py

```diff
@@ -265,9 +266,10 @@ def head_option(self):
 
         :return: first element of sequence or None if sequence is empty
         """
-        if not self.sequence:
+        try:
```

Comment:
ask forgiveness not permission
https://stackoverflow.com/a/12265860

Keeps from checking the if statement everytime